didcomm_resolver/resolver.py

```diff
@@ -89,32 +89,33 @@ async def update_connection(
         cls,
         session: ProfileSession,
         connection_id: str,
-        methods: Sequence[str],
+        methods: Set[str],
     ):
         """Update resolvers supported methods."""
         return await cls._upsert_methods(session, connection_id, methods, update=True)
 
     @classmethod
-    async def _upsert_methods(cls, session, connection_id, methods, update=False):
-
-        conn_record = await ConnRecord.retrieve_by_id(session, connection_id)
-        conn_record = cast(ConnRecord, conn_record)
-
-        if not update:
+    async def _upsert_methods(
+        cls, session, connection_id, methods: Set[str], update=False
+    ):
+        # retrieve connection record for metadata lookup, where methods are persisted
+        conn_record: ConnRecord = cast(
+            ConnRecord, await ConnRecord.retrieve_by_id(session, connection_id)
+        )
+        if update:
+            # retrieve current methods
             conn_record_metadata = await conn_record.metadata_get(
                 session, cls.METADATA_KEY
             )
-
+            # union methods
             if conn_record_metadata:
-                current_methods = conn_record_metadata.get(cls.METADATA_METHODS)
-                methods.extend(current_methods)
-
-        # Remove possible duplicates methods
-        methods = list(set(methods))
-
+                current_methods: Set = conn_record_metadata.get(cls.METADATA_METHODS)
+                methods = current_methods | methods
+        # update metadata records with new supported methods
```

Comment:
Before committing the changes to the methods. It is required to check if the methods are requested are not duplicated.

Reply:
with methods being sets, there should not be any duplicates. I use the pipe(|) here as a set union to update methods with new added methods. I think I see the issue that causes those tests to fail.

Reply:
Yep, it works. 
However in the case of updating the values. In a request with the following methods `["sov", "key", "sov"]`... There is not any current flow to validate the input of the request. I guess we can not assume all the possible errors from the input side, however, the union to delete duplicities can be done at the end, before changing anything in the connection.